packages/server/src/fhir/search.ts

```diff
@@ -905,33 +903,33 @@ function buildChainedSearch(selectQuery: SelectQuery, resourceType: string, para
   }
 }
 
-function buildSearchLinkCondition(
-  resourceType: string,
+function linkNextTable(
+  selectQuery: SelectQuery,
   link: ChainedSearchLink,
-  currentTable: string,
-  nextTable: string
-): Expression {
-  const linkColumn = new Column(currentTable, link.details.columnName);
-  if (link.reverse) {
-    const nextColumn = new Column(nextTable, link.details.columnName);
-    const currentColumn = new Column(currentTable, 'id');
+  currentResourceType: string,
+  currentTable: string
+): string {
+  const lookupAlias = selectQuery.getNextJoinAlias();
+  const nextTable = selectQuery.getNextJoinAlias();
+  const codeCondition = new Condition(new Column(lookupAlias, 'code'), '=', link.code);
 
-    if (link.details.array) {
-      return new ArraySubquery(
-        nextColumn,
-        new Condition(new Column(undefined, link.details.columnName), 'REVERSE_LINK', currentColumn, resourceType)
-      );
-    } else {
-      return new Condition(nextColumn, 'REVERSE_LINK', currentColumn, resourceType);
-    }
-  } else if (link.details.array) {
-    return new ArraySubquery(
-      linkColumn,
-      new Condition(new Column(nextTable, 'id'), 'LINK', new Column(undefined, link.details.columnName))
-    );
+  if (link.reverse) {
+    const lookupJoin = new Condition(new Column(currentTable, 'id'), '=', new Column(lookupAlias, 'targetId'));
```

Comment:
Oof.  This is intense.  It looks good, but intense.

Reply:
Intense probably means it should have more explanation, I'll add some comments!